File: packages/games_api/lib/src/models/player_stats.dart
import 'dart:convert';
import 'user.dart';

/// Represents player statistics for a specific game
class PlayerStats {
  final int id;
  final int userId;
  final int gameId;
  final String stats;
  final DateTime createdAt;
  final DateTime updatedAt;
  final User? user;

  PlayerStats({
    required this.id,
    required this.userId,
    required this.gameId,
    required this.stats,
    required this.createdAt,
    required this.updatedAt,
    this.user,
  });

  factory PlayerStats.fromJson(Map<String, dynamic> json) {
    return PlayerStats(
      id: json['id'] ?? 0,
      userId: json['user_id'] ?? 0,
      gameId: json['game_id'] ?? 0,
      stats: json['stats'] ?? '{}',
      createdAt: DateTime.parse(json['created_at'] ?? DateTime.now().toIso8601String()),
      updatedAt: DateTime.parse(json['updated_at'] ?? DateTime.now().toIso8601String()),
      user: json['user'] != null ? User.fromJson(json['user']) : null,
    );
  }

  /// Get stats data as a Map
  Map<String, dynamic> get statsData {
    try {
      return jsonDecode(stats);
    } catch (e) {
      return {};
    }
  }
}

File: api/app/models/player_stats.go
package models

import (
	"base/core/app/profile"
	"time"

	"gorm.io/gorm"
)

// PlayerStats stores player statistics per game
type PlayerStats struct {
	Id        uint           `gorm:"column:id;primary_key;auto_increment" json:"id"`
	UserId    uint           `gorm:"column:user_id;not null;index" json:"user_id" validate:"required"`
	User      *profile.User  `json:"user,omitempty" gorm:"foreignKey:UserId"`
	GameId    uint           `gorm:"column:game_id;not null;index" json:"game_id" validate:"required"`
	Game      *Game          `json:"game,omitempty" gorm:"foreignKey:GameId"`
	Stats     string         `gorm:"column:stats;type:json" json:"stats"` // JSON for scores, playtime, wins, etc.
	CreatedAt time.Time      `gorm:"column:created_at" json:"created_at"`
	UpdatedAt time.Time      `gorm:"column:updated_at" json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"column:deleted_at;index" json:"-"`
}

func (PlayerStats) TableName() string {
	return "player_stats"
}
